runelite-client/src/main/java/net/runelite/client/plugins/gauntlet/GauntletOverlay.java

```diff
@@ -0,0 +1,347 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.gauntlet;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.Graphics2D;
+import java.awt.Polygon;
+import java.awt.Rectangle;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.Model;
+import net.runelite.api.NPC;
+import net.runelite.api.Perspective;
+import net.runelite.api.Point;
+import net.runelite.api.Projectile;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.model.Jarvis;
+import net.runelite.api.model.Vertex;
+import net.runelite.client.graphics.ModelOutlineRenderer;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.OverlayUtil;
+import static net.runelite.client.util.ImageUtil.resizeImage;
+
+public class GauntletOverlay extends Overlay
+{
+	private static final int MAX_DISTANCE = 2400;
+	private final Client client;
+	private final GauntletPlugin plugin;
+	private final ModelOutlineRenderer outlineRenderer;
+
+	@Inject
+	private GauntletOverlay(Client client, GauntletPlugin plugin, ModelOutlineRenderer outlineRenderer)
+	{
+		this.client = client;
+		this.plugin = plugin;
+		this.outlineRenderer = outlineRenderer;
+
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.HIGH);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		// Save resources. There's nothing to render if the user is not in a raid.
+		if (!plugin.startedGauntlet())
+		{
+			return null;
+		}
+
+		if (plugin.fightingBoss())
+		{
+			// This section handles the visuals when the player is in the boss room.
+			// This section handles the projectile overlays.
+			Set<Missiles> projectiles = plugin.getProjectiles();
+			projectiles.forEach(projectile ->
+			{
+				BufferedImage icon = resizeImage(projectile.getImage(), plugin.getIconSize(), plugin.getIconSize());
+				Color color = projectile.getColor();
+
+				if (icon == null)
```

Comment:
This can never happen

This if statement can be removed (can't make multiline suggestions on Github sorry :( )